Core/MIPS/IR/IRCompBranch.cpp

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) 2012- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#include "profiler/profiler.h"
+
+#include "Core/Reporting.h"
+#include "Core/Config.h"
+#include "Core/MemMap.h"
+#include "Core/HLE/HLE.h"
+#include "Core/HLE/HLETables.h"
+
+#include "Core/MIPS/MIPS.h"
+#include "Core/MIPS/MIPSCodeUtils.h"
+#include "Core/MIPS/MIPSAnalyst.h"
+#include "Core/MIPS/MIPSTables.h"
+
+#include "Core/MIPS/IR/IRJit.h"
+#include "Core/MIPS/JitCommon/JitBlockCache.h"
+
+#include "Common/Arm64Emitter.h"
+
+#define _RS MIPS_GET_RS(op)
+#define _RT MIPS_GET_RT(op)
+#define _RD MIPS_GET_RD(op)
+#define _FS MIPS_GET_FS(op)
+#define _FT MIPS_GET_FT(op)
+#define _FD MIPS_GET_FD(op)
+#define _SA MIPS_GET_SA(op)
+#define _POS  ((op>> 6) & 0x1F)
+#define _SIZE ((op>>11) & 0x1F)
+#define _IMM16 (signed short)(op & 0xFFFF)
+#define _IMM26 (op & 0x03FFFFFF)
+
+#define LOOPOPTIMIZATION 0
+
+using namespace MIPSAnalyst;
+
+namespace MIPSComp
+{
+	using namespace Arm64Gen;
+
+void IRJit::BranchRSRTComp(MIPSOpcode op, IRComparison cc, bool likely)
+{
+	if (js.inDelaySlot) {
+		ERROR_LOG_REPORT(JIT, "Branch in RSRTComp delay slot at %08x in block starting at %08x", GetCompilerPC(), js.blockStart);
+		return;
+	}
+	int offset = _IMM16 << 2;
+	MIPSGPReg rt = _RT;
+	MIPSGPReg rs = _RS;
+	u32 targetAddr = GetCompilerPC() + offset + 4;
+
+	MIPSOpcode delaySlotOp = GetOffsetInstruction(1);
+	bool delaySlotIsNice = IsDelaySlotNiceReg(op, delaySlotOp, rt, rs);
+
+	ir.Write(IROp::Downcount, 0, js.downcountAmount & 0xFF, js.downcountAmount >> 8);
+
+	MIPSGPReg lhs = rs;
+	MIPSGPReg rhs = rt;
+	if (!delaySlotIsNice) {  // if likely, we don't need this
+		if (rs != 0) {
+			ir.Write(IROp::Mov, IRTEMP_LHS, rs);
+			lhs = (MIPSGPReg)IRTEMP_LHS;
+		}
+		if (rt != 0) {
+			ir.Write(IROp::Mov, IRTEMP_RHS, rt);
+			rhs = (MIPSGPReg)IRTEMP_RHS;
+		}
+	}
+
+	if (!likely)
+		CompileDelaySlot();
+
+	FlushAll();
```

Comment:
Hmm, so I guess these flushalls (at this point) don't actually do anything, right?  Otherwise we would somehow want to flush all EXCEPT IRTEMP_LHS/IRTEMP_RLHS.

-[Unknown]